### PR TITLE
feat(widget): use custom sounds in widget

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

1. Connect a wallet, setup a trade (not sign and send)
2. In widget configuration, in Customization section input a sound URL in "Submitted order sound URL" input. For example: https://swap.cow.fi/audio/send-winterTheme.mp3
3. Sign and send the trade. Wait until trade get executed
- [ ] ER: you hear winter sound instead of regular one
4. Check other sounds

image

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/sounds/utils/sound.ts [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+const COW_SOUND_TO_WIDGET_KEY: Record<SoundType, WidgetSounds> = {
+  SEND: 'postOrder',
+  SUCCESS: 'orderExecuted',
+  ERROR: 'orderError',
+}
+
 const SOUND_CACHE: Record<string, HTMLAudioElement | undefined> = {}
 
+function getWidgetSoundUrl(type: SoundType): string | undefined {
+  const { params } = jotaiStore.get(injectedWidgetParamsAtom)
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/sounds/utils/sound.ts:26]: Hacky? Yes
Does it work? Also yes

**Review by anxolin — APPROVED** [review]
It worked as a charm

Tested with:
https://cdn.pixabay.com/download/audio/2023/04/11/audio_300c872ea4.mp3?filename=fart-145914.mp3

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @shoom3301 , great!
Tried to add mp3, wav and ogg links to files, all work. 

But it would be great to add to docs info about accepted file formats and links, otherwise users will see this in the console: 
invalid url

Also, I'd add a possibility of removing a sound completely. Currently, when the field is empty, it uses CoW Swap default sounds. 

And I have not been able to test cases with failed onchain TXs as they are not working in dev/PRs right now. 

Thanks!

**Comment by shoom3301** [comment]
>Also, I'd add a possibility of removing a sound completely

Nice idea, could you please create a task for it?

**Pull request merged**